roachpb/data_test.go

```diff
@@ -385,6 +386,50 @@ func TestNodeList(t *testing.T) {
 	}
 }
 
+func TestTransactionUpdate(t *testing.T) {
+	nodes := NodeList{
+		Nodes: []int32{101, 103, 105},
+	}
+	ts := makeTS(10, 11)
+
+	txn := Transaction{
+		Name:          "name",
+		Key:           Key("foo"),
+		ID:            uuid.NewUUID4(),
+		Priority:      957356782,
+		Isolation:     SNAPSHOT,
+		Status:        COMMITTED,
+		Epoch:         2,
+		LastHeartbeat: &ts,
+		Timestamp:     makeTS(20, 21),
+		OrigTimestamp: makeTS(30, 31),
+		MaxTimestamp:  makeTS(40, 41),
+		CertainNodes:  nodes,
+		Writing:       true,
+	}
+
+	noZeroField := func(txn Transaction) bool {
+		ele := reflect.ValueOf(&txn).Elem()
+		for i := 0; i < ele.NumField(); i++ {
+			f := ele.Field(i)
+			zero := reflect.Zero(f.Type())
+			if reflect.DeepEqual(f.Interface(), zero.Interface()) {
+				return false
```

Comment:
it's going to be more helpful for the test if you return an `error` instead of a `boolean`, then you can print the offending field.